src/app/features/course-info/course-info.component.ts

```diff
@@ -1,10 +1,44 @@
-import { Component } from '@angular/core';
-
+import { Component, Input, OnInit } from "@angular/core";
+import { CoursesService } from "@app/services/courses.service";
+interface Course {
+  id: string;
+  title: string;
+  description: string;
+  creationDate: string;
+  duration: number;
+  authors: string[];
+}
 @Component({
-  selector: 'app-course-info',
-  templateUrl: './course-info.component.html',
-  styleUrls: ['./course-info.component.scss']
+  selector: "app-course-info",
+  templateUrl: "./course-info.component.html",
+  styleUrls: ["./course-info.component.scss"],
 })
-export class CourseInfoComponent {
+export class CourseInfoComponent implements OnInit {
+  @Input() course!: Course;
   // Use the names for the input `course`.
+  id: string = "";
+  title: string = "";
+  description: string = "";
+  creationDate: string = "";
+  duration: number = 0;
+  authors: string[] = [];
+
+  constructor(private coursesService: CoursesService) {}
```

Comment:
As of now, this is unused. Until CoursesService is put to use, you can delete it from the constructor.